From 605c3dc996865ed684f42b2772dbf370fb20a12c Mon Sep 17 00:00:00 2001
From: Gianluca Scopelliti <gianluca.scopelliti@ericsson.com>
Date: Mon, 6 May 2024 16:15:58 +0000
Subject: [PATCH] common.sh: allow checkout from commit based on date

---
 common.sh | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/common.sh b/common.sh
index 2bbab60..c8f9b1d 100755
--- a/common.sh
+++ b/common.sh
@@ -10,6 +10,17 @@ run_cmd()
 	}
 }
 
+checkout_repo() {
+	git restore .
+	BRANCH=$1
+	run_cmd git checkout current/${BRANCH}
+
+	if [ -n "$LATEST_COMMIT_DATE" ];then
+		COMMIT=$(git log --until $LATEST_COMMIT_DATE | grep commit | head -n 1 | awk '{ print $2 }')
+		run_cmd git checkout $COMMIT
+	fi
+}
+
 build_kernel()
 {
 	set -x
@@ -68,7 +79,7 @@ build_kernel()
 
 		pushd ${V} >/dev/null
 			run_cmd git fetch current
-			run_cmd git checkout current/${BRANCH}
+			checkout_repo $BRANCH
 			COMMIT=$(git log --format="%h" -1 HEAD)
 
 			run_cmd "cp /boot/config-$(uname -r) .config"
@@ -175,7 +186,7 @@ build_install_ovmf()
 
 	pushd ovmf >/dev/null
 		run_cmd git fetch current
-		run_cmd git checkout current/${OVMF_BRANCH}
+		checkout_repo $OVMF_BRANCH
 		run_cmd git submodule update --init --recursive
 		run_cmd make -C BaseTools
 		. ./edksetup.sh --reconfig
@@ -214,7 +225,7 @@ build_install_qemu()
 
 	pushd qemu >/dev/null
 		run_cmd git fetch current
-		run_cmd git checkout current/${QEMU_BRANCH}
+		checkout_repo $QEMU_BRANCH
 		run_cmd ./configure --target-list=x86_64-softmmu --prefix=$DEST
 		run_cmd $MAKE
 		run_cmd $MAKE install
-- 
2.34.1

